platform/linux-generic/pktio/pcap.c

```diff
@@ -188,15 +188,18 @@ static int pcapif_promisc_mode_set(pktio_entry_t *pktio_entry,
 
 	if (pcap_compile(pcap->rx, &bpf, filter_exp,
 			 0, PCAP_NETMASK_UNKNOWN) != 0) {
+		pcap_freecode(&bpf);
```

Comment:
When pcap_compile fails, bpf is not written and pcap_freecode(&bpf) should not be called.